jgrapht-core/src/main/java/org/jgrapht/GraphTests.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * (C) Copyright 2016-2016, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.jgrapht.graph.AbstractBaseGraph;
+
+/**
+ * A collection of utilities to test for various graph properties.
+ * 
+ * @since September 2016
+ */
+public abstract class GraphTests
+{
+
+    /**
+     * Check if a graph is simple, i.e. has no self-loops and multiple edges.
+     * 
+     * @param graph a graph
+     * @param <V> the graph vertex type
+     * @param <E> the graph edge type
+     * @return true if a graph is simple, false otherwise
+     */
+    public static <V, E> boolean isSimple(Graph<V, E> graph)
+    {
+        if (graph == null) {
+            throw new IllegalArgumentException("Graph cannot be null");
+        }
+        if (graph instanceof AbstractBaseGraph<?, ?>) {
+            AbstractBaseGraph<V, E> abg = (AbstractBaseGraph<V, E>) graph;
+            if (!abg.isAllowingLoops() && !abg.isAllowingMultipleEdges()) {
+                return true;
+            }
+        }
+        // no luck, we have to check
+        boolean isDirected = graph instanceof DirectedGraph<?, ?>;
+        for (V v : graph.vertexSet()) {
+            Set<V> neighbors = new HashSet<>();
+            Set<E> edgesOf;
+            if (isDirected) {
+                edgesOf = ((DirectedGraph<V, E>) graph).outgoingEdgesOf(v);
+            } else {
+                edgesOf = graph.edgesOf(v);
+            }
+            for (E e : edgesOf) {
+                V u = Graphs.getOppositeVertex(graph, e, v);
+                if (u.equals(v)) {
+                    return false;
+                }
+                if (!neighbors.add(u)) {
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+}
```

Comment:
What about the other tests that are present in the current experimental GraphTests? Do we want to keep those, or did you leave them out on purpose? If you left them out on purpose, we may want to rethink having a dedicated GraphTests class, unless you envision more tests to be added somewhere in the future.